servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/DefaultExecutor.java

```diff
@@ -249,11 +254,27 @@ public Cancellable schedule(final Runnable task, final long delay, final TimeUni
             // When using the global scheduler, offload timer ticks to the user specified Executor since user code
             // executed on the timer tick can block.
             ScheduledFuture<?> future = GLOBAL_SINGLE_THREADED_SCHEDULED_EXECUTOR.schedule(
-                    () -> offloadExecutor.execute(task), delay, unit);
-            // Schedulers are only used to generate a tick and do not execute any user code. This means they will never
-            // run any blocking code and hence it does not matter whether we use the interruptOnCancel as sent by the
-            // user upon creation in the scheduler. User code (completion of Completable on tick) will be executed on
-            // the configured executor and not the Scheduler thread.
+                    () -> {
+                        try {
+                            offloadExecutor.execute(task);
+                        } catch (Throwable t) {
+                            LOGGER.error("Unexpected exception while offloading scheduled task: {} to executor: {}.",
+                                    task, offloadExecutor, t);
+                            // If the offloadExecutor failed to execute, we fallback to the scheduler thread for
+                            // best-effort attempt at actually executing the task.
+                            try {
+                                task.run();
+                            } catch (Throwable taskFailure) {
+                                LOGGER.error("Scheduled task {} threw exception during fallback to scheduler executor.",
+                                        task, taskFailure);
+                            }
+                        }
+                    }, delay, unit);
+            // Schedulers are only used to generate a tick and should not execute any user code (unless the
+            // offloadExecutor throws). This means they will never run any blocking code and hence it does not matter
```

Comment:
Yes, this is not ideal that we can still run a blocking code on this thread, but should be ok as a workaround for now. In future, we probably need to expend `schedule` API to also take a consumer of an error.